### PR TITLE
pkp/pkp-lib#3071 Fix review form use counts (and clean up related code)

### PR DESCRIPTION
**Pull request by asmecher**

### CODE REVIEW

File: controllers/grid/settings/reviewForms/ReviewFormGridCellProvider.inc.php [hunk 1]
```diff
@@ -29,24 +29,15 @@ function getTemplateVarsFromRowColumn($row, $column) {
...
+				return array('label' => $element->getCompleteCount());
 			case 'active':
-				$selected = $element->getActive();
-				return array('selected' => $selected);
-				break;
-			default:
-				break;
+				return array('selected' => $element->getActive());
 		}
+		return parent::getTemplateVarsFromRowColumn($row, $column);
```
Comment (NateWr) [inline controllers/grid/settings/reviewForms/ReviewFormGridCellProvider.inc.php]: This is a minor nitpick, but if this were ever sub-classed, or if we ever wanted `GridCellProvider::getTemplateVarsFromRowColumn` to return a global configuration parameter, the parent method would get skipped with the `returns` in the `switch`.

My preference would be to merge the results of the parent with the child in cases like this:

```
switch ($columnId) {
  case 'name':
    $vars = array('label' => 'whatever');
  ...
}
$parentVars = parent::getTemplateVarsFromRowColumns($row, $column);
return array_merge($parentVars, $vars);
```

File: classes/reviewForm/ReviewFormDAO.inc.php [hunk 2]
```diff
@@ -35,20 +35,12 @@ function getById($reviewFormId, $assocType = null, $assocId = null) {
...
 			FROM	review_forms rf
-				LEFT JOIN review_assignments rac ON (
-					rac.review_form_id = rf.review_form_id AND
-					rac.date_confirmed IS NOT NULL
-				)
-				LEFT JOIN review_assignments rai ON (
-					rai.review_form_id = rf.review_form_id AND
-					rai.date_notified IS NOT NULL AND
-					rai.date_confirmed IS NULL
-				)
```
[the remark this answers is not in the document]
Comment (NateWr) [reply classes/reviewForm/ReviewFormDAO.inc.php]: Yeah I just tested and a declined review counts as "In Review" in the grid. I think the same issue is in the `getByAssocId` function below.

[the remark this answers is not in the document]
Comment (asmecher) [reply classes/reviewForm/ReviewFormDAO.inc.php]: I thought about this too -- the usage counts are used both for display (in which arguably declined reviews should be excluded) and in checking whether a form can be deleted (in which arguably they shouldn't). This is consistent with earlier intended behavior and I'm OK to wait for a specific requirement before refining it further. I think these are basically used at a glance for an editor to know "OK, this is the form I've been using most", but I'm guessing.

File: controllers/grid/settings/reviewForms/ReviewFormGridCellProvider.inc.php (hunk 1 above)
Comment (asmecher) [reply controllers/grid/settings/reviewForms/ReviewFormGridCellProvider.inc.php]: That is a bigger convention change, but it's a good idea. However, best not dealt with in this context. I have some plans for heavier lifting in parent classes etc. and that would be a good time to introduce this.

**Pull request merged**